examples/llm/src/models/mosaic_gpt.py

```diff
@@ -279,9 +320,21 @@ def prepare_inputs_for_generation(self,
         if past_key_values is not None:
             input_ids = input_ids[:, -1].unsqueeze(-1)
 
+        if self.prefix_lm:
+            # Leverage a convenience of sequential generation!
+            prefix_mask = torch.ones_like(attention_mask)
+            # This requires that we're using the cache
+            if kwargs.get('use_cache') == False:
```

Comment:
why is it that cache is required for this?

Reply:
In order to use the trick that prefix_mask=1 everywhere, it needs to be the case that previous tokens (i.e. tokens that were supplied as the input or already generated) do not have their key/value embeddings updated. Otherwise, this trick would cause those embeddings to change each step, which would be inconsistent with how the model was trained.

Reply:
ahh got it, thanks!